### ISSUE TITLE
Incorrect type when spreading two Sets to an object

### ISSUE THREAD

**Issue by styfle** [microsoft/TypeScript#47999]
# Bug Report

I had a typo where I used curly brackets instead of square brackets to spread to Set variables, but TypeScript didn't catch the error at build time. Instead, the code failed at runtime.

### 🔎 Search Terms

set, object, spread

### 🕗 Version & Regression Information

- This is the behavior in every version I tried back to TS 3.x

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?target=8&module=1#code/MYewdgzgLgBAZgSwE7RgXhmApgdxgZSygAoBtAcnC3IF0BKAKFElgi2YBN1NcCizyUHCFqNmqAIYAbKdwDeMAHTLEKKABolytpxgBfANxNwEEFKyKpIAObFpUujAD0TmAAckCMFAgwABnJ6fgz2ihIcHMTkOBJQ5I4uMAAq+DAIvgAWEm5uAJ4wAEYArrBQGUggOL4SMEhF3ggAtlgwWEgVSEA)

### 💻 Code

```ts
const first = new Set(['one'])
const second = new Set(['two'])
const all = { ...first, ...second };
console.log(all) // prints `{}`
all.add('wat') // TS is happy but JS throws a runtime error
```

### 🙁 Actual behavior

TS thinks the Set methods like `.add()` and `.clear()` are available and doesn't show a build error for the last line.

### 🙂 Expected behavior

TS should show an error at build time to reflect actual JS usage at runtime: no Set method are available for the last line.

**Comment by nmain** [comment]
I believe this is a duplicate of https://github.com/microsoft/TypeScript/issues/9726

**Comment by styfle** [comment]
Yep, looks like a very similar to #33081 which is a duplicate of #9726
